File: src/main/java/no/nav/sbl/util/AuthContextUtils.java
package no.nav.sbl.util;

import com.nimbusds.jwt.JWTClaimsSet;
import lombok.SneakyThrows;
import no.nav.common.auth.context.AuthContextHolder;
import no.nav.common.utils.StringUtils;
import java.util.Optional;

public class AuthContextUtils {
    public static final String AAD_NAV_IDENT_CLAIM = "NAVident";
    public static String requireIdToken() {
        return AuthContextHolder.requireIdTokenString();
    }

    public static Optional<String> getIdent() {
        return AuthContextHolder.getIdTokenClaims()
                .map(AuthContextUtils::getSubject)
                .filter(StringUtils::notNullOrEmpty);
    }

    @SneakyThrows
    private static String getSubject(JWTClaimsSet claims) {
        String navIdent = claims.getStringClaim(AAD_NAV_IDENT_CLAIM);
        return navIdent != null ? navIdent : claims.getSubject();
    }
}

File: src/main/java/no/nav/sbl/util/AuthContextUtils.kt
package no.nav.sbl.util

import com.nimbusds.jwt.JWTClaimsSet
import no.nav.common.auth.context.AuthContext
import no.nav.common.auth.context.AuthContextHolderThreadLocal
import no.nav.common.utils.fn.UnsafeRunnable
import no.nav.sbl.util.AccesstokenServletFilter.AccesstokenHolderThreadLocal
import java.util.*

object AuthContextUtils {
    private val authContextHolder = AuthContextHolderThreadLocal.instance()
    private val accesstokenHolder = AccesstokenHolderThreadLocal.instance()

    @JvmStatic
    fun getAccessToken(): Optional<String> = accesstokenHolder.getAccesstoken()

    @JvmStatic
    fun getIdTokenClaims(): Optional<JWTClaimsSet> = authContextHolder.idTokenClaims

    @JvmStatic
    fun requireIdTokenString(): String = authContextHolder.requireIdTokenString()

    @JvmStatic
    fun withContext(authContext: AuthContext?, block: UnsafeRunnable) = authContextHolder.withContext(authContext, block)

    @JvmStatic
    fun withAccesstoken(accesstoken: String?, block: UnsafeRunnable) = accesstokenHolder.withAccesstoken(accesstoken, block)
}